flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -585,7 +586,7 @@ public void testStatusMetrics() throws Exception {
         // sleep a bit to ensure uptime is > 0
         Thread.sleep(10L);
 
-        assertThat(upTimeGauge.getValue()).isGreaterThan(0L);
+        CommonTestUtils.waitUntilCondition(() -> upTimeGauge.getValue() > 0L);
```

Comment:
The `Thread.sleep(10L)` can be removed if we add the waitUntil condition. Additionally, the same pattern with `Thread.sleep` and `isGreatherThan` is applied a few lines further down in the test code. We might want to update that as well.